packages/runner/injection/multi-domain.js

```diff
@@ -9,16 +9,29 @@
 
 import { createTimers } from './timers'
 
-// TODO: don't hard-code the index. need it to be predictable or need
-// to search for the right one somehow. will need to be fixed when we
-// test out visiting a 3rd domain
-const cyBridgeFrame = window.parent.frames[2]
-
-const Cypress = cyBridgeFrame.Cypress
+let Cypress
+
+for (let index = 0; index < window.parent.frames.length; index++) {
+  const frame = window.parent.frames[index]
+
+  try {
+    // If Cypress is defined and we haven't gotten a cross domain error we have found the correct bridge.
+    if (frame.Cypress) {
+      Cypress = frame.Cypress
+    }
```

Comment:
Maybe `break` after this. Once we've found the right frame, there should be no need to keep looping through the frames.

Reply:
updated